### DISCUSSION TITLE
Full download exception: Expecting value: line 1 column 1 (char 0)

### DISCUSSION THREAD

**Discussion by onursoyaslan** [Michael-K-Stein/SpotiFile#4] [Q&A]
I get the following error:
"2023-04-06 21:33:09 | Full download exception: Expecting value: line 1 column 1 (char 0)"

How can i resolve it?

<img width="1791" alt="image" src="https://user-images.githubusercontent.com/121290672/230465832-132af97d-f746-4abc-a369-064ff8c710bc.png">

**Comment by Michael-K-Stein** [comment]
Try removing the question mark and everything after it from the url

**Reply by Ed-o** [reply]
I get same error without the question mark and trailing text.  All types of searches
only thing I see in the console output is : Access token is anon:  False
and nothing downloads at all

**Comment by ronspe** [comment]
@Ed-o: I had the same problem with "Acces token is anon" what worked for me was logging out and back in on Spotify.

I also had an error: "Failed to load globals save file! Exception: [Errno 2] No such file or directory: '_downloaded_store.json'" 
That was solved by changing the file permissions of that file and download folder.
Hope this helps...

In the meantime, I can also not download because of the " Full download exception: Expecting value: line 1 column 1 (char 0)" error. This is the url I tried: https://open.spotify.com/playlist/4YewL89lNrCIsaIYJiwVHr

**Comment by mikedhanson** [comment]
Running into the same problem

**Comment by sasquelch** [comment]
Specifically, there is an error with parsing JSON during the call to get the client token:

File "/Users/sasquelch/Documents/SpotiFile/spotify_client.py", line 60, in get_client_token return response.json()['granted_token']['token'] ^^^^^^^^^^^^^^^ File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/requests/models.py", line 975, in json raise RequestsJSONDecodeError(e.msg, e.doc, e.pos) requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

**Comment by ronspe** [comment]
I noticed something strange.  @sasquelch mentioned the spotify_client.py line 60.
When I debug and set a breakpoint on line 60 I can see my client_id. But when I clicked "download" in the GUI the client_id becomes "unknown". But why is that happening?
![image](https://user-images.githubusercontent.com/44189214/230716155-76ac563b-6804-454e-8e0f-1ff7641c7b2d.png)

**Comment by sasquelch** [comment]
I'm starting to think that @Michael-K-Stein never had this code working and is looking for a way to get the community to find a fix for him. ;)
